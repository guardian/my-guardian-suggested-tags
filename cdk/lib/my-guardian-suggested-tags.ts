import type { GuStackProps } from "@guardian/cdk/lib/constructs/core";
import { GuStack } from "@guardian/cdk/lib/constructs/core";
import { GuS3Bucket } from "@guardian/cdk/lib/constructs/s3";
import type { App } from "aws-cdk-lib";
import { StringParameter } from "aws-cdk-lib/aws-ssm";

export class MyGuardianSuggestedTags extends GuStack {
  constructor(scope: App, id: string, props: GuStackProps) {
    super(scope, id, props);
    const bucket = new GuS3Bucket(this, `${props.app}`, {
      app: props.app as string,
    });

    new StringParameter(this, "MyGuardianSuggestedTagsBucketName", {
      description: "The name of the bucket where the site is hosted",
      parameterName: `/${props.stage}/${props.stack}/${props.app}/bucket-name`,
      stringValue: bucket.bucketName,
    });
  }
}
