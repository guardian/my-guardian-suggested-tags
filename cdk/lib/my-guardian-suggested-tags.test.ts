import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { MyGuardianSuggestedTags } from "./my-guardian-suggested-tags";

describe("The MyGuardianSuggestedTags stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const stack = new MyGuardianSuggestedTags(app, "MyGuardianSuggestedTags", { stack: "mobile", stage: "TEST", app: "my-guardian-suggested-tags-static" },);
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
