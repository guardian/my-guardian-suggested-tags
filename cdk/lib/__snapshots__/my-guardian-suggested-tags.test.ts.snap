// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MyGuardianSuggestedTags stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [],
    "gu:cdk:version": "TEST",
  },
}
`;
